### ISSUE TITLE
Loop throug features only

### ISSUE THREAD

**Issue by JulienM28** [rfennell/AzurePipelines#631]
I Added a custom input in my process:
Custom.Details

I want to loop on work items having his custom input, no matter his value, so I did:

@@WILOOP[ALL]:Custom.Details@@
${Custom.Details}
@@WILOOP@@

But it's nor working.

Can I do that ?

Thanks

**Comment by rfennell** [comment]
You would need to use the field filtering option
```
@@WILOOP[ALL]:System.WorkItemType=Bug@@

```

Let me know how it goes

**Comment by JulienM28** [comment]
@@WILOOP[ALL]:System.WorkItemType=Bug@@ works.
I found the field making a dump of all fields.

I can iterate on my features for example.

But as I added a custom field : Custom.ReleaseNote, this field exist only on the features where it' filled. So when I launch the release note generator I have an error saying it's not defined.

That's why I wonder if I can filter on the existence of the field in the object, and not his value.

Thanks.


.

**Comment by rfennell** [comment]
Interesting idea, enhance the ANY|ALL logic to have some form of null operator. That will require a code change so I have swapped to an enhancement label from question.

It should be doable but I need to have a look into it. Not sure how soon that will be

**Comment by JulienM28** [comment]
Thank you.

**Comment by rfennell** [comment]
This has been released as 2.25.1. There is now a wildcard option for field matching. In effect it just checks the field exists on the WI, it is not bothered by content.

The default wildcard character is *, but you can change this in the advanced settings

```
### Should be same as generic loop with no filters as every WI has a title
@@WILOOP[ANY]:System.Title=*@@  
* **${widetail.fields['System.WorkItemType']} ${widetail.id}**   ${widetail.fields['System.Title']}  
@@WILOOP[ANY]:System.Title=*@@  

### Should be empty as the field not present on any WI
@@WILOOP[ANY]:Custom.Title=*@@  
* **${widetail.fields['System.WorkItemType']} ${widetail.id}**   ${widetail.fields['System.Title']}  
@@WILOOP[ANY]:Custom.Title=*@@  
```
